src/modules/adaptors/Sommelier/ERC4626Adaptor.sol

```diff
@@ -0,0 +1,175 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.21;
+
+import { BaseAdaptor, ERC20, SafeTransferLib, Cellar } from "src/modules/adaptors/BaseAdaptor.sol";
+import { ERC4626 } from "@solmate/mixins/ERC4626.sol";
+
+/**
+ * @title Generic ERC4626 Vault Adaptor (basically a copy of Cellar Adaptor w/ virtual function sigs to enable inheritance and overriding).
+ * @notice Allows Cellars to interact with other ERC4626 contracts.
+ * @author crispymangoes, 0xEinCodes
+ * @dev this is a WIP and not audited yet.
+ * NOTE: `CellarAdaptor.sol` was used, and is still used for nested Cellar positions for Sommelier Cellars (ERC4626 Vaults). Integrations into Aura, among other ERC4626 contracts outside of Sommelier have led to using a separate ERC4626Vault.sol file that allows more flexibility via inheritance.
+ * NOTE: `Cellar` is still used throughout this contract at times because it is a contract made for the Sommelier protocol (has extra pricing and accounting mechanics, etc.)
+ */
+contract ERC4626Adaptor is BaseAdaptor {
```

Comment:
I think this contract should be moved to be with the `ERC20Adaptor.sol`, since its not a somm specific contract, but interested to hear your thoughts

Reply:
When you say "be with the `ERC20Adaptor.sol`." Do you mean it's file location? If so, yes I agree.

Reply:
I've moved the file to where `ERC20Adaptor.sol` is stored within the repo.